packages/main/src/DynamicDateRange.ts

```diff
@@ -0,0 +1,250 @@
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import jsxRenderer from "@ui5/webcomponents-base/dist/renderer/JsxRenderer.js";
+import query from "@ui5/webcomponents-base/dist/decorators/query.js";
+import i18n from "@ui5/webcomponents-base/dist/decorators/i18n.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { isDesktop } from "@ui5/webcomponents-base/dist/Device.js";
+import "@ui5/webcomponents-icons/dist/appointment-2.js";
+import DynamicDateRangeTemplate from "./DynamicDateRangeTemplate.js";
+import IconMode from "./types/IconMode.js";
+import type Input from "./Input.js";
+import type { IDynamicDateRangeOption } from "./DynamicDateOption.js";
+import {
+	DYNAMIC_DATE_RANGE_SELECTED_TEXT,
+	DYNAMIC_DATE_RANGE_EMPTY_SELECTED_TEXT,
+} from "./generated/i18n/i18n-defaults.js";
+
+// default calendar for bundling
+import "@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js";
+
+// Styles
+import dynamicDateRangeCss from "./generated/themes/DynamicDateRange.css.js";
+import dynamicDateRangePopoverCss from "./generated/themes/DynamicDateRangePopover.css.js";
+import ResponsivePopoverCommonCss from "./generated/themes/ResponsivePopoverCommon.css.js";
+
+type DynamicDateRangeValue = {
+	/**
+	 * The key of the option.
+	 * @default ""
+	 * @public
+	 */
+	operator: string;
+
+    /**
+     * Values of the dynamic date range.
+     * @default []
+     * @public
+     */
+    values?: Date[] | number[];
+}
+
+/**
+ * @class
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents/dist/DynamicDateRange.js";`
+ * @constructor
+ * @extends UI5Element
+ * @public
+ */
+
+@customElement({
+	tag: "ui5-dynamic-date-range",
+	languageAware: true,
+	template: DynamicDateRangeTemplate,
+	renderer: jsxRenderer,
+	styles: [
+		dynamicDateRangeCss,
+		ResponsivePopoverCommonCss,
+		dynamicDateRangePopoverCss,
+	],
+})
+
+class DynamicDateRange extends UI5Element {
+    @i18n("@ui5/webcomponents")
+	static i18nBundle: I18nBundle;
+
+	/**
+	 * Defines a formatted date value.
+	 * @default undefined
+	 * @public
+	 */
+	@property({ noAttribute: true })
+	value?: DynamicDateRangeValue;
+
+	/**
+	 * Defines the open or closed state of the popover.
+	 * @public
```

Comment:
Should this be public? Do we have a case where the app will pre-open the popup?

Reply:
Usually we have them public. But having them public would require events, because we internally change the state of the property. I think we can have it as private until we receive a request for it.